tests/Model/AgentTest.php

```diff
@@ -33,4 +41,131 @@ public function testCanAccess() {
+                'temps' => json_encode(array("0" => ["09:00:00","12:30:00","13:30:00","17:00:00","4"],"1" => ["09:00:00","12:30:00","13:30:00","17:00:00","4"],"2" => ["09:00:00","12:30:00","13:30:00","17:00:00","4"],"3" => ["08:00:00","12:30:00","13:30:00","17:00:00","4"],"4" => ["08:00:00","12:30:00","13:30:00","17:00:00","4"]))
+            )
+        );
+        $this->assertEquals($agent2->getWorkingHoursOn($date2->format('Y-m-d'))['temps'][0][0], '09:00:00','les heures sont bien enregistrées');
+        $this->assertEquals($agent2->getWorkingHoursOn($date2->format('Y-m-d'))['temps'][0][2], '13:30:00','les heures sont bien enregistrées');
+    }
+
+    public function testSkills(){
+        global $entityManager;
+        $builder = new FixtureBuilder();
+        $builder->delete(Agent::class);
+
+        $date = \DateTime::createFromFormat("d/m/Y", date('d/m/Y'));
+        $start = \DateTime::createFromFormat("H:i:s", '00:01:00');
+        $end = \DateTime::createFromFormat("H:i:s", '23:59:00');
+
+        $skill1 = $builder->build(Skill::class, array('nom' => 'basket'));
+        $id1 = $skill1->id();
+        $skill2 = $builder->build(Skill::class, array('nom' => 'volley'));
+        $id2 = $skill2->id();
+        $agent = $builder->build(Agent::class, array('login' => 'jdevoe', 'postes' => json_encode(["$id1", "$id2"])));
+
+        $this->assertEquals($id1, $agent->skills()[0]);
+        $this->assertEquals($id2, $agent->skills()[1]);
+    }
+
+    public function test_get_planning_unit_mails(){
+        global $entityManager;
+        $builder = new FixtureBuilder();
+        $builder->delete(Agent::class);
+
+        $start = \DateTime::createFromFormat("H:i:s", '08:00:00');
+        $end = \DateTime::createFromFormat("H:i:s", '17:00:00');
+        $date = \DateTime::createFromFormat("d/m/Y", '09/10/2022');
+
+        $agent = $builder->build(Agent::class, array('login' => 'jdevoe', 'mail' => 'j.devoe@mail.com'));
+
+        $GLOBALS['config']['Mail-Planning'] = 'j.devoe@mail.com;jm@mail.fr;j.devoe@mail.com';
+
+        $this->assertEquals($agent->get_planning_unit_mails(),['j.devoe@mail.com','jm@mail.fr']);
+
+        //Plusieurs sites :
+
+        $GLOBALS['config']['Multisites-nombre'] = 4;
+        $GLOBALS['config']['Mail-Planning'] = '';
+
+        $agent2 = $builder->build(Agent::class, array('login' => 'jmarc', 'sites' => json_encode(["1", "2", "3","4"])));
+        $GLOBALS['config']['Multisites-site1-mail'] ='jmarc@mail.fr;jcharles@mail.fr';
+        $GLOBALS['config']['Multisites-site2-mail'] ='jcharles@mail.fr;jmarc@mail.fr;j.paul@mail.com';
+        $GLOBALS['config']['Multisites-site3-mail'] ='j.claude@mail.com;jmarc@mail.fr;jcharles@mail.fr';
+        $GLOBALS['config']['Multisites-site4-mail'] ='j.paul@mail.com;j.claude@mail.com';
+
+        $this->assertEquals($agent2->get_planning_unit_mails(),[0 => 'jmarc@mail.fr', 1 => 'jcharles@mail.fr', 4 => 'j.paul@mail.com', 5 => 'j.claude@mail.com']);
```

Comment:
Here i would like a test with an agent that is attached to only one site. And an other attached to site 1 and 3 (for exemple).